src/utils/format.rs

```diff
@@ -0,0 +1,28 @@
+use format_num::NumberFormat;
+
+const PATTERN: &str = ".3s";
+
+pub fn si_format<T>(num: T) -> String
+where
+  T: Into<f64>,
+{
+  let fmt = NumberFormat::new();
```

Comment:
Since this formatter is always the same, you could make this a static using `LazyLock`.